### ISSUE TITLE
Simple structural type does not link

### ISSUE THREAD

**Issue by propensive** [scala-native/scala-native#643]
The following code reports a `scala.scalanative.util.package$UnsupportedException` and prints a stack trace in Scala Native 0.1:

```scala
object Main {
  val foo: { def get: Int } = Some(42)
  def main(args: Array[String]): Unit = println(foo.get)
}
```

[Output from `sbt run`](https://github.com/scala-native/scala-native/files/930487/output.txt)

**Comment by densh** [comment]
Thanks for the bug report, @propensive !

**Comment by densh** [comment]
Fixed in #645
